### PR TITLE
bacon: added programs.bacon

### PR DESCRIPTION
**Pull request by shimunn**
bacon is a background rust code checker.

https://dystroy.org/bacon/

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all` or `nix develop --ignore-environment .#all` using Flakes.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [x] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

### CODE REVIEW

File: modules/programs/bacon.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      type = settingsFormat.type;
+      example = {
+        jobs.default = {
+          command = [ "cargo" "build" "--all-features" "--color" "always" ];
+          need_stdout = true;
+        };
+      };
+      description = ''
+        Bacon configuration.
+        for available settings see https://dystroy.org/bacon/#global-preferences
```
Comment (rycee) [inline modules/programs/bacon.nix:27]:
```suggestion
        For available settings see <https://dystroy.org/bacon/#global-preferences>.
```

File: modules/programs/bacon.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+  package = pkgs.bacon;
+  settingsFormat = pkgs.formats.toml { };
+in {
+  meta.maintainers = [ maintainers.shimunn ];
+
+  options.programs.bacon = {
+    enable = mkEnableOption "background rust code check";
+
+    preferences = mkOption {
```
Comment (rycee) [inline modules/programs/bacon.nix:17]: This option should be called `settings` as per RFC-42:
```suggestion
    settings = mkOption {
```

File: modules/programs/bacon.nix [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+with lib;
+
+let
+
+  cfg = config.programs.bacon;
+
+  package = pkgs.bacon;
+  settingsFormat = pkgs.formats.toml { };
+in {
+  meta.maintainers = [ maintainers.shimunn ];
```
Comment (rycee) [inline modules/programs/bacon.nix:12]:
```suggestion
  meta.maintainers = [ hm.maintainers.shimunn ];
```

File: modules/programs/bacon.nix [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+      };
+      description = ''
+        Bacon configuration.
+        for available settings see https://dystroy.org/bacon/#global-preferences
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    home.packages = [ package ];
```
Comment (rycee) [inline modules/programs/bacon.nix:33]: The `package` variable doesn't add much, can just inline it:

```suggestion
    home.packages = [ pkgs.bacon ];
```

File: modules/programs/bacon.nix [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+  cfg = config.programs.bacon;
+
+  package = pkgs.bacon;
+  settingsFormat = pkgs.formats.toml { };
+in {
+  meta.maintainers = [ maintainers.shimunn ];
+
+  options.programs.bacon = {
+    enable = mkEnableOption "background rust code check";
```
Comment (rycee) [inline modules/programs/bacon.nix:15]:
```suggestion
    enable = mkEnableOption "bacon, a background rust code checker";
```

**Comment by rycee** [comment]
Thanks for the contribution! I added a few comments. Also please consider adding a basic test case.

File: modules/programs/bacon.nix [hunk 6]
```diff
@@ -0,0 +1,38 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.bacon;
+
+  package = pkgs.bacon;
```
Comment (rycee) [inline modules/programs/bacon.nix:9]:
```suggestion
```

**Comment by rycee** [comment]
Looks good. About the test, it just need to be a simple assert that the output file is what you expect. See, for example, the comodoro test: https://github.com/nix-community/home-manager/tree/master/tests/modules/programs/comodoro

**Comment by shimunn** [comment]
@rycee all good?

**Comment by rycee** [comment]
Thanks! Merged to master now 🙂

**Pull request merged**